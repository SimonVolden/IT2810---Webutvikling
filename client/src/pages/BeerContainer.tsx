import React, { Fragment } from 'react';
import CardHeader from '@mui/material/CardHeader';
import Avatar from '@mui/material/Avatar';
import Card from '@mui/material/Card';
import { Beer } from '../interfaces/Beer';
import Typography from '@mui/material/Typography';
import BeerDescription from './BeerDescription';
import BeerMethods from './BeerMethods';
import BeerIngredients from './BeerIngredients';
import { useSelector } from 'react-redux';
import { AppState } from '../stateManagement/types';
import { createTheme, ThemeProvider } from '@mui/material';

interface BeerContainerProps {
    beer: Beer
}


function BeerContainer(props: BeerContainerProps): JSX.Element {
    
    
    const pageTheme = useSelector((state: AppState) => state.theme)
    const cardTheme = pageTheme ? "#424242" :"#ffffff"

    const theme = createTheme({
        palette: {
            mode: pageTheme ? "dark" : "light",
        }
    })


    return (
        <ThemeProvider theme={theme}>
        <Card sx={{ marginBottom: "1vh", }} square={true} elevation={3} style={{ backgroundColor: cardTheme }}>
            <CardHeader 
                avatar={
                    <Avatar alt={props.beer.name} src={props.beer.image_url} variant="square" sx={{ width: "25px !important", height: "auto !important" }} />
                }
                title={
                    <Typography variant="h6">{props.beer.name}</Typography>
                }
                subheader={
                    <Fragment>
                        <Typography variant="subtitle2">{props.beer.tagline}</Typography>
                        <Typography variant="subtitle1">{props.beer.abv +"%"}</Typography>
                    </Fragment>
                }
            />
            
                <BeerDescription name="Description" desc={props.beer.description} food_pairing={props.beer.food_pairing} />
                <BeerMethods name="Methods/Timings" method={props.beer.method} brewers_tips={props.beer.brewers_tips} />
                <BeerIngredients name="Ingredients" ingredients={props.beer.ingredients} />
            
        </Card>
        </ThemeProvider>
    );
}

export default BeerContainer;
