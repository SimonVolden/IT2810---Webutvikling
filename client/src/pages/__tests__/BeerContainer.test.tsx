import React from 'react';
import { render} from '@testing-library/react';
import BeerContainer from '../BeerContainer'

class Beer {
    id: number
    name: string;
    image_url: string;
    description: string; 
    tagline: string;
    likes: number

    constructor(id: number, name: string, 
        image_url: string, description: string, tagline: string, likes: number) {
            this.id = id;
            this.name = name;
            this.image_url = image_url;
            this.description = description;
            this.tagline = tagline;
            this.likes = likes
        }
}

it("renders", () => {
    const beer = new Beer(1,
        "Beer_test",
        "https://images.punkapi.com/v2/keg.png",
        "description",
        "tnaglie",
        1)
    const { container } = render(<BeerContainer beer={beer} key={beer.id} />);
    expect(container).toMatchSnapshot();
});