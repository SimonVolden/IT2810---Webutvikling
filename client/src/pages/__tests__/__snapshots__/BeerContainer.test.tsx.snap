// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <nav
    aria-label=""
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <img
        aria-label="picture of beer,"
        src="https://images.punkapi.com/v2/keg.png"
        style="width: 7%; height: auto; padding-right: 10px;"
      />
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <h6
          aria-label="beer name:, Beer_test , "
          class="MuiTypography-root MuiTypography-h6"
        >
          Beer_test
        </h6>
      </div>
    </li>
    <div
      class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        aria-label="Beer info:description , "
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Description
        </span>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        aria-label="Beer info:tnaglie , "
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Tagline
        </span>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
  </nav>
</div>
`;
